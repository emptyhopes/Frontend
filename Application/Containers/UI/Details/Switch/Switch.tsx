import styled from "styled-components";

import { PropsInterface } from "@/Application/Containers/UI/Details/Switch/Types/PropsInterface";

const StyledSwitch = styled.label`
  position: relative;
  display: inline-block;
  width: 40px;
  height: 20px;
  cursor: pointer;
`;

const StyledInput = styled.input`
  width: 0;
  height: 0;
  opacity: 0;

  &:focus + span {
    box-shadow: 0 0 1px #2196f3;
  }

  &:checked + span {
    background-color: #2196f3;
  }

  &:checked + span::before {
    transform: translateX(18px);
  }
`;

const StyledSpan = styled.span`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 34px;
  background-color: #cccccc;
  cursor: pointer;
  transition: 0.2s;

  &::before {
    position: absolute;
    bottom: 3px;
    left: 4px;
    width: 14px;
    height: 14px;
    border-radius: 50%;
    background-color: #ffffff;
    content: "";
    transition: 0.2s;
  }
`;

const Switch: React.FunctionComponent<PropsInterface> = (props: PropsInterface): React.ReactElement => {
  return (
    <StyledSwitch>
      <StyledInput type="checkbox" {...props} />
      <StyledSpan />
    </StyledSwitch>
  );
};

export { Switch };
