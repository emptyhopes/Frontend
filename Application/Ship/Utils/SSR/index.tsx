import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
