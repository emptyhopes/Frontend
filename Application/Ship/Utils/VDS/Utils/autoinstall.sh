#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
