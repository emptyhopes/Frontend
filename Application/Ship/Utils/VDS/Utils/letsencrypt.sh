#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname