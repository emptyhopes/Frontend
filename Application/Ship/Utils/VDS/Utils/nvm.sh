#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
