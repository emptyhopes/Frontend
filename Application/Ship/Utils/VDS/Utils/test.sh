#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru