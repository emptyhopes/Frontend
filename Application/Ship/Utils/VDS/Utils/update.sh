#! /bin/bash

sudo apt-get update --yes && sudo apt-get upgrade --yes
sudo apt-get autoremove --yes

sudo apt-get install git nginx certbot python3-certbot-nginx --yes