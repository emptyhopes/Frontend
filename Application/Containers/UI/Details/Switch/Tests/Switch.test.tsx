import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Switch } from "@/Application/Containers/UI/Details/Switch/Switch";

test("Switch", () => {
  const { container } = RenderTests(<Switch />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
