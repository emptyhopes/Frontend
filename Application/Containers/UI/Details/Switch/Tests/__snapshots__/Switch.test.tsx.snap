// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 40px;
  height: 20px;
  cursor: pointer;
}

.c1 {
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:focus + span {
  box-shadow: 0 0 1px #2196f3;
}

.c1:checked + span {
  background-color: #2196f3;
}

.c1:checked + span::before {
  -webkit-transform: translateX(18px);
  -ms-transform: translateX(18px);
  transform: translateX(18px);
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 34px;
  background-color: #cccccc;
  cursor: pointer;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c2::before {
  position: absolute;
  bottom: 3px;
  left: 4px;
  width: 14px;
  height: 14px;
  border-radius: 50%;
  background-color: #ffffff;
  content: "";
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

<div>
  <label
    class="c0"
  >
    <input
      aria-label="switch"
      class="c1"
      type="checkbox"
    />
    <span
      class="c2"
    />
  </label>
</div>
`;
